### ISSUE TITLE
No difference between `allow when using the app` and `always`

### ISSUE THREAD

**Issue by jdevogelaere-rmdy** [Baseflow/flutter-permission-handler#345]
## 🐛 Bug Report

When request the locationAlways permission status. It returns granted even when the user selects 'allow while using'. This was the main breaking change according location services since iOS 13 one year ago.

### Expected behavior
Please make a distinction between these 2 permission levels.

### Reproduction steps
- Request location permissions
- Select while using (since you can't select always since iOS 13)
- request status ( `final permissions = await Permission.locationAlways.status;`)
- It returns granted.
- Navigate to the settings app. There you see the user selected "while using"
### Configuration

**Version:** 5.0.1+1

**Platform:** 
- [x] :iphone: iOS 13

**Comment by ruben4mobile** [comment]
Experiencing the same issue.

Requesting two permissions:

```
    var statuses = await [
      Permission.locationAlways,
      Platform.isIOS ? Permission.sensors : Permission.activityRecognition
    ].request();
```

But it always shows the option to 
**Allow while Using App**
**Allow once**
**Don't Allow**

**Comment by jdevogelaere-rmdy** [comment]
@ruben4mobile the options that you see are correct. Since iOS 13 you can't give **Allow always** authorization immediately (You need to do this manually in your settings, or the OS fires randomly the pop up to access your location**always.**. The issue is that when your request the permission status there is no difference between **always** and **while using**.

**Comment by ruben4mobile** [comment]
@jdevogelaere-rmdy Thanks, for the quick response.
I am also using another plugin and in a blog post I found that it is also impossible to read the always permission status.

So I guess we have to live with this crapy decision of apple.
I seems to get only worse with iOS 14 and extra permission
https://hypertrack.com/blog/2020/06/24/impact-of-ios13-ios14-location-permissions-on-background-location-access/

**Comment by LongXiangGuo** [comment]
- the PermissionStatus is missing one of the kCLAuthorizationStatus  
see https://github.com/Baseflow/flutter-permission-handler/blob/eb7ef50a35be6023948147c08178844a317430d5/permission_handler/ios/Classes/strategies/LocationPermissionStrategy.m#L126

```Objective-C
        switch (authorizationStatus) {
            case kCLAuthorizationStatusNotDetermined:
                return PermissionStatusNotDetermined;
            case kCLAuthorizationStatusRestricted:
                return PermissionStatusRestricted;
            case kCLAuthorizationStatusDenied:
                return PermissionStatusDenied;
            case kCLAuthorizationStatusAuthorizedWhenInUse: 
            case kCLAuthorizationStatusAuthorizedAlways:
                return PermissionStatusGranted;
        }
```

**Comment by JDDV** [comment]
@LongXiangGuo the PermissionStatus is not missing, it should return `PermissionStatusGranted`.
@ruben4mobile & @jdevogelaere-rmdy today a new version of the plugin is released (version ^8.1.4+1). This version fixed some location bugs which should also solve the problems you are facing.

There are however a few changes since the issue mentioned version ^5.0.1+1. You need to update the `podfile` as mentioned in the [documentation](https://github.com/Baseflow/flutter-permission-handler/blob/master/permission_handler/README.md#setup).

Also, when requesting the `locationAlways` permission you need to first request the `locationWhenInUse` permission. After allowing the `locationWhenInUse` permission by clicking on `Allow When Using App` you can request the `locationAlways` permission.

I'll close this issue for now, but please reopen the issue if you still have questions!
